### PR TITLE
[16.0][MIG] sale_order_general_discount_triple

### PR DESCRIPTION
**Pull request by RabbitJon-S73**
Standard migration to 16.0

Depends on https://github.com/OCA/sale-workflow/pull/2318

### CODE REVIEW

**Comment by rousseldenis** [comment]
@RabbitJon-S73 Don't forget to write an understandable PR's title (not the default one provided by github based on your branch name)

**Comment by rousseldenis** [comment]
@RabbitJon-S73 Could you also add in your PR description the depending PR(s) ?

You could also follow this guide to include it here to help reviewers testing on runboat (and make tests passing).

https://github.com/OCA/maintainer-tools/wiki/Use-temporary-reference%28s%29-to-another-pull-request%28s%29

**Comment by RabbitJon-S73** [comment]
> @RabbitJon-S73 Could you also add in your PR description the depending PR(s) ?
> 
> You could also follow this guide to include it here to help reviewers testing on runboat (and make tests passing).
> 
> https://github.com/OCA/maintainer-tools/wiki/Use-temporary-reference%28s%29-to-another-pull-request%28s%29

Updated!

**Comment by RabbitJon-S73** [comment]
@pedrobaeza could you review?

**Comment by RabbitJon-S73** [comment]
@rousseldenis  @pedrobaeza

File: sale_order_general_discount_triple/models/sale_order_line.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        digits="Discount",
+        compute=False,
+        store=True,
+        readonly=False,
+        default=0.0,
+    )
+
+    @api.model
+    def default_get(self, fields):
+        vals = super(SaleOrderLine, self).default_get(fields)
```
Comment (rousseldenis) [inline sale_order_general_discount_triple/models/sale_order_line.py]:
```suggestion
        vals = super().default_get(fields)
```

**Review by rousseldenis — APPROVED** [review]
Code review. Minor comment

File: sale_order_general_discount_triple/models/sale_order_line.py (hunk 1 above)
Comment (RabbitJon-S73) [reply sale_order_general_discount_triple/models/sale_order_line.py]: @rousseldenis Done!

**Review by ioans73 — APPROVED** [review]
LGTM 👍🏻

**Review by rousseldenis — APPROVED** [review]
Code review

**Comment by RabbitJon-S73** [comment]
Hi @rousseldenis could we merge this PR?

**Comment by pedrobaeza** [comment]
Changing the `discount` field to a non computed one is provoking all sort of errors when the module is installed, as the standard flows have been totally screwed up. I don't know the reason for that, but you should find another technique for such goal. I have directly disabled it in 9aadac128a383c3883605a381edf34577e806f72 while you find the approach for not provoking errors on CIs. Splitting the CIs isolating the module is not a solution, as anyone using the module will experiment the problem, so we will only trick the system, not solve the problem.

**Comment by RabbitJon-S73** [comment]
> Changing the `discount` field to a non computed one is provoking all sort of errors when the module is installed, as the standard flows have been totally screwed up. I don't know the reason for that, but you should find another technique for such goal. I have directly disabled it in [9aadac1](https://github.com/OCA/sale-workflow/commit/9aadac128a383c3883605a381edf34577e806f72) while you find the approach for not provoking errors on CIs. Splitting the CIs isolating the module is not a solution, as anyone using the module will experiment the problem, so we will only trick the system, not solve the problem.

https://github.com/OCA/sale-workflow/pull/2392 here I revert the non-compute, but there are others changes done. Should I split it in two PR?

**Comment by pedrobaeza** [comment]
Well, a comprehensive separation of commits with the explanation of the changes are the way to go. If all fit in the same commit, but with a good explanation why they are not split, it's OK as well.

**Pull request merged**